scripts/SREP/retry-pruning-cronjob/script.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+# Delete jobs that fail in openshift-sre-pruning namespace.
+
+set -x
+set -e
+set -o nounset
+set -o pipefail
+
+readonly PRUNING_NS=openshift-sre-pruning
+FailedJobs=()
+
+detect_job() {
+  echo "INFO: finding jobs producing error"
+  for current_ns in ${PRUNING_NS}; do
+    FailedJobs+=($(oc get jobs -n "$current_ns" -o=jsonpath='{.items[?(@.status.failed>0)].metadata.name}'))
+  done
+}
+
+delete_job() {
+  if [[ ${#FailedJobs[@]} == 0 ]]; then
+    echo "INFO: no failed jobs found exiting.."
+    return 1
+  else
+    echo "INFO: deleting jobs producing error"
+    for jobs in "${FailedJobs[@]}"; do
+      oc delete job "$jobs" -n "$PRUNING_NS"
+      FailedJobs=("${FailedJobs[@]:1}")
+    done
+  fi
+}
+
+retry_job() {
+  detect_job
```

Comment:
This function looks redundant since it is doing the same thing as the main function.

From the main() function: 
```
main() {
  detect_job
  delete_job
  retry_job
  echo "Succeed."
  exit 0
}
```
From the execution order above, looks you detect and delete failed jobs, then call `retry_job`, which again detects and deletes failed jobs. However, there is **no actual retrying of the jobs happening**,  you are just deleting them?